File: backend/WholesaleERP.API/Models/Item.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WholesaleERP.API.Models
{
    public class Item
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; } = string.Empty;

        [StringLength(1000)]
        public string? Description { get; set; }

        [Required]
        [StringLength(100)]
        public string SKU { get; set; } = string.Empty;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Quantity { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal CostPrice { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal SalePrice { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal LowStockThreshold { get; set; } = 10;

        [StringLength(50)]
        public string? Unit { get; set; } // kg, pieces, liters, etc.

        public bool IsActive { get; set; } = true;

        [StringLength(500)]
        public string? ImageUrl { get; set; }

        // Foreign Key
        public int? CategoryId { get; set; }

        // Navigation Properties
        public virtual Category? Category { get; set; }
        public virtual ICollection<InvoiceItem> InvoiceItems { get; set; } = new List<InvoiceItem>();
        public virtual ICollection<PurchaseItem> PurchaseItems { get; set; } = new List<PurchaseItem>();

        // Audit fields
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Computed properties
        [NotMapped]
        public bool IsLowStock => Quantity <= LowStockThreshold;

        [NotMapped]
        public decimal ProfitMargin => CostPrice == 0 ? 0 : ((SalePrice - CostPrice) / CostPrice) * 100;
    }
}

File: backend/src/entities/Item.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, OneToMany, JoinColumn } from 'typeorm';
import { Category } from './Category';
import { InvoiceItem } from './InvoiceItem';
import { PurchaseItem } from './PurchaseItem';

@Entity('items')
export class Item {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'varchar', length: 255 })
  name: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column({ type: 'varchar', length: 100, unique: true })
  sku: string;

  @Column({ type: 'decimal', precision: 10, scale: 2, default: 0 })
  quantity: number;

  @Column({ type: 'decimal', precision: 10, scale: 2 })
  costPrice: number;

  @Column({ type: 'decimal', precision: 10, scale: 2 })
  salePrice: number;

  @Column({ type: 'decimal', precision: 10, scale: 2, default: 10 })
  lowStockThreshold: number;

  @Column({ type: 'varchar', length: 50, nullable: true })
  unit: string; // kg, pieces, liters, etc.

  @Column({ type: 'boolean', default: true })
  isActive: boolean;

  @Column({ type: 'varchar', length: 255, nullable: true })
  imageUrl: string;

  @ManyToOne(() => Category, category => category.items, { nullable: true })
  @JoinColumn({ name: 'categoryId' })
  category: Category;

  @Column({ nullable: true })
  categoryId: number;

  @OneToMany(() => InvoiceItem, invoiceItem => invoiceItem.item)
  invoiceItems: InvoiceItem[];

  @OneToMany(() => PurchaseItem, purchaseItem => purchaseItem.item)
  purchaseItems: PurchaseItem[];

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  // Virtual property to check if stock is low
  get isLowStock(): boolean {
    return this.quantity <= this.lowStockThreshold;
  }

  // Virtual property to calculate profit margin
  get profitMargin(): number {
    if (this.costPrice === 0) return 0;
    return ((this.salePrice - this.costPrice) / this.costPrice) * 100;
  }
}